esx_service/vmdk_ops.py

```diff
@@ -844,24 +848,48 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
     except ValidationError as ex:
         return err(str(ex))
 
-    if not datastore:
-        datastore = default_datastore
-
     if datastore and not vmdk_utils.validate_datastore(datastore):
         return err("Invalid datastore '%s'.\n" \
                    "Known datastores: %s.\n" \
                    "Default datastore: %s" \
                    % (datastore, ", ".join(get_datastore_names_list()), default_datastore))
 
```

Comment:
No, unless ALL_DS is set for the tenant. For DEFAULT tenant,  since ALL_DS is set post install, so it will be able to create volumes on any datastore to make sure it backward compatible. For named tenant, privilege to ALL_DS is not created unless user add this privilege explicitly.